### PR TITLE
Reduced split-k range

### PR DESCRIPTION
**Pull request by djramic**
This PR builds upon the work done in  #1613.
There are changes to the split-k range, now using only 1 & 4 during tuning.

### CODE REVIEW

**Comment by manupak** [comment]
Can you also share the perf numbers for the configs that worked ?

File: mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp [hunk 1]
```diff
@@ -129,21 +129,16 @@ computeOptimalSplitKFactors(GemmSize origGemmSize, int32_t gemmMPerBlock,
...
-        computeWorkImbalance(origGemmSize, gemmMPerBlock, gemmNPerBlock,
-                             gemmKPerBlock, kPack, numCUs, splitKFactor);
-    const auto gain = dataPrallelGemmImbalance / imbalance;
-    if (gain > minGain) {
-      factors.emplace_back(LocalData{splitKFactor, imbalance});
-    }
+  // A large set of splitK values significantly increases tuning time,
+  // after analysis, we've determined that using only splitK factor 4 is
+  // sufficient.
+  constexpr int32_t splitKFactor = 4;
```
Comment (krzysz00) [inline mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp]: I'm not sure about this - I know from back in the old `atomic_add` implicit split-K stuff in backwards weight that we had a heuristic for picking between, say, 4, 8, or 16, or something vaguely like that

Comment (manupak) [reply mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp]: this was Ethan's hypothesis... but I d let perf numbers decide that.

**Comment by djramic** [comment]
> Can you also share the perf numbers for the configs that worked ?

Here are the performance results for all GEMM and convolution configurations that are working:
[gfx90a_mlir_vs_miopen_perf.csv](https://github.com/user-attachments/files/16640718/gfx90a_mlir_vs_miopen_perf.csv)
[gfx90a_mlir_vs_rocblas_perf.csv](https://github.com/user-attachments/files/16640720/gfx90a_mlir_vs_rocblas_perf.csv)

Comparison with tuning perfconfigs db from CI:
[Conv comparison.xlsx](https://github.com/user-attachments/files/16641422/Conv.comparison.xlsx)
[Gemm comparison.xlsx](https://github.com/user-attachments/files/16641424/Gemm.comparison.xlsx)

**Comment by djramic** [comment]
Ticked [#1577](https://github.com/ROCm/rocMLIR-internal/issues/1577) describes the issue that arose during conv tuning with enabled splitK

**Comment by dhernandez0** [comment]
I got performance numbers of the whole range (https://github.com/ROCm/rocMLIR/pull/1613) vs this PR. For convolutions I didn't see any relevant change. However, for GEMM I got these cases where the speed up was smaller than 0.95. However, the TFlops are small, maybe it could be due to noise?

MI200 f16 full range:
```csv
,DataType,OutDataType,Chip,numCU,TransA,TransB,G,M,K,N,PerfConfig,LDSBankConflict,TFlops
1,f16,f16,gfx90a,110,False,False,1,2,1280,1280,"v2:16,16,8,16,16,4,25,1,1",4.483529279299999,0.514572864321608
2,f16,f16,gfx90a,110,False,False,1,2,2816,1280,"v2:16,64,8,16,16,4,8,1,1",0.28072899298,0.831291512915129
3,f16,f16,gfx90a,110,False,False,1,2,320,1280,"v2:16,16,4,16,16,4,21,1,1",0.75680104233,0.1585139318885449
```

MI200 f16 reduced range:
```csv
,DataType,OutDataType,Chip,numCU,TransA,TransB,G,M,K,N,PerfConfig,LDSBankConflict,TFlops
1,f16,f16,gfx90a,110,False,False,1,2,1280,1280,"v2:32,32,8,16,16,8,4,1,1",1.0051557546999998,0.4785046728971962
2,f16,f16,gfx90a,110,False,False,1,2,2816,1280,"v2:16,64,8,16,16,8,4,1,1",0.8049409616300001,0.8017081850533806
3,f16,f16,gfx90a,110,False,False,1,2,320,1280,"v2:16,16,8,16,16,8,4,1,1",0.26204407573,0.14104683195592285
```

**Comment by manupak** [comment]
This seems good enough evidence to reduce the split-k range to [1,4]. Did you try the configs in https://github.com/ROCm/rocMLIR-internal/issues/1421 ?

**Comment by manupak** [comment]
(also while you are here please feel free update the sdxl-conv-configs with configs in https://github.com/ROCm/rocMLIR-internal/issues/1421)

**Comment by dhernandez0** [comment]
> This seems good enough evidence to reduce the split-k range to [1,4]. Did you try the configs in [ROCm/rocMLIR-internal#1421](https://github.com/ROCm/rocMLIR-internal/issues/1421) ?

yes, I tried those, also "sdxl-conv" and "sdxl-gemm" in the repo. I'm running the same thing for f32 on MI200 and f16 on Navi3

**Comment by manupak** [comment]
Can you double check whether they (https://github.com/ROCm/rocMLIR-internal/issues/1421) pick split-k at all in MI200 ? If not I ll have to check this with MI300

**Comment by dhernandez0** [comment]
> Can you double check whether they ([ROCm/rocMLIR-internal#1421](https://github.com/ROCm/rocMLIR-internal/issues/1421)) pick split-k at all in MI200 ? If not I ll have to check this with MI300

yes, some pick split-k 2 for the full range and 4 for the reduced range:

full range:
```csv
,Direction,DataType,Chip,numCU,FilterLayout,InputLayout,OutputLayout,N,C,H,W,K,Y,X,DilationH,DilationW,StrideH,StrideW,PaddingH,PaddingW,PerfConfig,LDSBankConflict,TFlops
0,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1280,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,2,1,1",0.69286767555,111.63537924954115
1,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1280,64,64,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",0.79028983662,126.01269161377276
2,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1280,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",0.73715629428,118.7207170656917
3,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1920,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,128,32,8,2,1,1",0.47083330377,117.46953132496064
4,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1920,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,64,32,8,1,1,1",0.4736894942,118.900132225049
5,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,2560,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,2,1,1",0.35818031749,115.03752196309343
6,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,128,128,320,3,3,1,1,1,1,1,1,"v2:64,128,8,32,32,8,1,1,1",2.95343520154,117.13638040343584
7,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,128,128,320,3,3,1,1,2,2,1,1,"v2:64,64,8,32,32,8,1,1,1",2.63747434294,100.37355101606695
8,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,128,128,4,3,3,1,1,1,1,1,1,"v2:32,16,8,32,16,8,1,1,1",13.1117676973,5.960781638598735
9,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,4,128,16,8,1,1,1",30.82345601472,123.21591578603778
10,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,4,128,128,320,3,3,1,1,1,1,1,1,"v2:64,128,4,32,16,4,1,1,1",7.926256165800001,19.71007518796992
11,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,128,128,320,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.61200249438,107.90497898104988
12,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,128,128,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.61246801626,128.1409971570416
13,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,2,1,1",1.3075631916200001,105.56060430226299
14,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.44188913786,116.82484506649955
15,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,64,64,640,3,3,1,1,2,2,1,1,"v2:64,64,8,32,32,8,1,1,1",1.2857578966000003,98.52915451324314
16,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,960,128,128,320,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.1697099860400002,120.10566789140546
17,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,960,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.0665128231999998,132.45073705601567
```

reduced range:
```csv
,Direction,DataType,Chip,numCU,FilterLayout,InputLayout,OutputLayout,N,C,H,W,K,Y,X,DilationH,DilationW,StrideH,StrideW,PaddingH,PaddingW,PerfConfig,LDSBankConflict,TFlops
0,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1280,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,4,1,1",1.4069460376699998,113.0870379940234
1,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1280,64,64,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",0.79053539186,126.09924823332483
2,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1280,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",0.73574325958,118.56038066223292
3,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1920,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,4,1,1",0.9385512972400001,116.60831730880614
4,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,1920,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",0.4692294026,117.22625930255803
5,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,2560,32,32,1280,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,4,1,1",0.7530595135499999,120.38664062188559
6,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,128,128,320,3,3,1,1,1,1,1,1,"v2:64,128,8,64,32,8,1,1,1",2.96109893828,117.33321729129918
7,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,128,128,320,3,3,1,1,2,2,1,1,"v2:64,64,8,32,32,8,1,1,1",2.6528529461,100.54465330909527
8,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,128,128,4,3,3,1,1,1,1,1,1,"v2:128,64,8,128,16,8,1,1,1",12.912186631500001,5.8864211699946205
9,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,4,128,16,8,1,1,1",30.81626880638,122.81503436495994
10,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,4,128,128,320,3,3,1,1,1,1,1,1,"v2:64,128,4,16,16,4,1,1,1",7.2120909462,17.271566617862373
11,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,128,128,320,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.60980069588,107.72349378250539
12,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,128,128,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.61279101888,128.25868689710825
13,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,32,32,1280,3,3,1,1,1,1,1,1,"v2:64,64,8,32,32,8,1,1,1",1.41279952386,104.20846808422534
14,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.44248925856,116.91168149404194
15,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,640,64,64,640,3,3,1,1,2,2,1,1,"v2:64,64,8,32,32,8,1,1,1",1.2937452051,98.94235857649286
16,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,960,128,128,320,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.16965336584,120.11076333211142
17,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,960,64,64,640,3,3,1,1,1,1,1,1,"v2:128,128,8,32,32,8,1,1,1",1.0636833662199998,132.0861510468834
```

**Comment by dhernandez0** [comment]
For Navi3 no perfConf picks split-k for "sdxl-conv" and "sdxl-gemm" in the repo or https://github.com/ROCm/rocMLIR-internal/issues/1421

**Comment by manupak** [comment]
Please check other configs in the repo as well

**Comment by dhernandez0** [comment]
> Please check other configs in the repo as well

I've found these two experiments in "conv-configs" where there's are speed ups 0.93 and 0.81. The second one is a case where performance is clearly worse. In both cases splitk=9.

full range:
```csv
,Direction,DataType,Chip,numCU,FilterLayout,InputLayout,OutputLayout,N,C,H,W,K,Y,X,DilationH,DilationW,StrideH,StrideW,PaddingH,PaddingW,PerfConfig,LDSBankConflict,TFlops
32,fwd,f32,gfx90a,110,k01c,n01c,n01k,2,320,64,64,4,3,3,1,1,1,1,1,1,"v2:16,128,8,16,16,4,9,1,1",6.143364043850001,3.7906426735218512
72,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,64,64,4,3,3,1,1,1,1,1,1,"v2:16,128,8,16,16,8,9,1,1",5.53622755364,7.080720288115246
```

reduced range:
```csv
,Direction,DataType,Chip,numCU,FilterLayout,InputLayout,OutputLayout,N,C,H,W,K,Y,X,DilationH,DilationW,StrideH,StrideW,PaddingH,PaddingW,PerfConfig,LDSBankConflict,TFlops
32,fwd,f32,gfx90a,110,k01c,n01c,n01k,2,320,64,64,4,3,3,1,1,1,1,1,1,"v2:16,64,8,16,16,8,4,1,1",8.149525247880002,3.5297666068222617
72,fwd,f16,gfx90a,110,k01c,n01c,n01k,2,320,64,64,4,3,3,1,1,1,1,1,1,"v2:16,64,8,16,16,8,4,1,1",6.47651347371,5.771272015655576
```

**Comment by manupak** [comment]
I think split-k = 8, should give a similiar perf as well.

What is interesting is that doubling the split-k allows us to have a 128 nPerBlock. Could you check the gridsize in these configs ?
 
My hypothesis is if split-k allows more CU utilization whilst keeping dPerBlock for some d at 128 thats beneficial. In the same time, the K dimension should be significant after the split to amortize atomic contention. Here the K dimension = 320 * 3 * 3. 

Ok I see why 9 is interesting... coz that d mean there d be 320 per split.
So kPerBlock = 8 * 4  = 32, therefore there would be 10 iterations of K.

**Comment by dhernandez0** [comment]
> I think split-k = 8, should give a similiar perf as well.
> 
> What is interesting is that doubling the split-k allows us to have a 128 nPerBlock. Could you check the gridsize in these configs ?
> 
> My hypothesis is if split-k allows more CU utilization whilst keeping dPerBlock for some d at 128 thats beneficial. In the same time, the K dimension should be significant after the split to amortize atomic contention. Here the K dimension = 320 * 3 * 3.
> 
> Ok I see why 9 is interesting... coz that d mean there d be 320 per split. So kPerBlock = 8 * 4 = 32, therefore there would be 10 iterations of K.

For the splitK=4 perfConf, grid_size = 512 and block_size = 256.
For the splitK=9 perfConf, grid_size = 576 and block_size = 256.

**Comment by manupak** [comment]
As discussed lets move "finding the better heuristic" part to a seperate ticket and get this in.
@krzysz00 WDYT ?

**Comment by dhernandez0** [comment]
I found some more cases of performance regression with the reduced range for gemm on MI200.

f16 full range:
```csv
,DataType,OutDataType,Chip,numCU,TransA,TransB,G,M,K,N,PerfConfig,LDSBankConflict,TFlops
16,f16,f16,gfx90a,110,False,False,1,128,5120,1280,"v2:64,128,4,64,32,8,8,1,1",0.8364827525999999,49.50783758262511
18,f16,f16,gfx90a,110,False,False,2,64,1280,3840,"v2:64,128,4,32,16,8,5,1,1",1.9747563120599998,38.893768545994064
28,f16,f16,gfx90a,110,False,False,1,1,1024,1024,"v2:32,64,4,32,16,4,8,1,1",0.49597409882,0.17569973190348526
41,f16,f16,gfx90a,110,False,False,1,64,768,768,"v2:16,64,8,16,16,8,6,1,1",0.97429868297,6.51739226519337
```

f16 reduced range:
```csv
,DataType,OutDataType,Chip,numCU,TransA,TransB,G,M,K,N,PerfConfig,LDSBankConflict,TFlops
16,f16,f16,gfx90a,110,False,False,1,128,5120,1280,"v2:32,64,8,16,16,8,4,1,1",3.2542142618600005,46.07100175746923
18,f16,f16,gfx90a,110,False,False,2,64,1280,3840,"v2:32,128,4,32,32,8,4,1,1",0.61667799974,35.91013698630137
28,f16,f16,gfx90a,110,False,False,1,1,1024,1024,"v2:16,64,8,16,16,8,4,1,1",0.35227736498000006,0.15868280871670704
41,f16,f16,gfx90a,110,False,False,1,64,768,768,"v2:64,16,4,16,16,8,4,1,1",1.6635015000300002,5.883531172069826
```

f32 full range:
```csv
,DataType,OutDataType,Chip,numCU,TransA,TransB,G,M,K,N,PerfConfig,LDSBankConflict,TFlops
21,f32,f32,gfx90a,110,False,False,1,384,768,2,"v2:64,16,8,16,16,8,6,1,1",0.45453353224,0.10654335260115606
23,f32,f32,gfx90a,110,False,False,1,384,3072,768,"v2:64,64,4,32,16,4,11,1,1",5.795754869990001,26.397717482517482
26,f32,f32,gfx90a,110,False,False,1,1,768,768,"v2:16,32,4,16,16,4,10,1,1",0.6570986759,0.10017391304347825
28,f32,f32,gfx90a,110,False,False,1,1,1024,1024,"v2:16,32,4,16,16,4,11,1,1",1.03462992541,0.1567846889952153
```

f32 reduced range:
```csv
,DataType,OutDataType,Chip,numCU,TransA,TransB,G,M,K,N,PerfConfig,LDSBankConflict,TFlops
21,f32,f32,gfx90a,110,False,False,1,384,768,2,"v2:16,64,8,16,16,8,4,1,1",2.1566752217999996,0.08653521126760562
23,f32,f32,gfx90a,110,False,False,1,384,3072,768,"v2:64,64,8,32,32,4,4,1,1",0.14528739072,24.597017959682344
26,f32,f32,gfx90a,110,False,False,1,1,768,768,"v2:64,16,8,16,16,8,4,1,1",2.9629993668,0.08048908296943232
28,f32,f32,gfx90a,110,False,False,1,1,1024,1024,"v2:32,32,8,16,16,4,4,1,1",1.1366710145,0.13739203354297694
```

**Comment by manupak** [comment]
Ack... please note these down in the future ticket. Still I dont think its worthwhile tuning for them.. we should be looking at a heuristic instead.

File: mlir/lib/Dialect/Rock/Generator/ConvGenerator.cpp [hunk 2]
```diff
@@ -841,6 +842,13 @@ LogicalResult ConvGenerator::genConvModule(ModuleOp &module, int rawKernelId,
   // Construct the FuncOp.
   func = func::FuncOp::create(builder.getUnknownLoc(), kernelName, funcType,
                               ArrayRef<NamedAttribute>(kernelAttrs));
+  // TODO[split-K]: remove after integrating split-K into MIGraphX
```
Comment (manupak) [inline mlir/lib/Dialect/Rock/Generator/ConvGenerator.cpp]: Please remove the comment.

File: mlir/include/mlir/Dialect/Rock/Generator/ConvGenerator.h [hunk 3]
```diff
@@ -32,6 +32,8 @@ class ConvGenerator {
     std::string arch;
     // TODO: drop these
     std::string chip;
+    // TODO[split-K]: remove after integrating split-K into MIGraphX
```
Comment (manupak) [inline mlir/include/mlir/Dialect/Rock/Generator/ConvGenerator.h]: Please remove the comment

**Review by manupak — APPROVED** [review]
LGTM based on the numbers, I think its not worthwhile increasing the tuning space to hit the few regressions we are seeing. Rather, we should look at a heuristic.

Also `disableSplitKForTuning` is a useful handle to have;  I dont see the need for remove it.

**Comment by dhernandez0** [comment]
> Ack... please note these down in the future ticket. Still I dont think its worthwhile tuning for them.. we should be looking at a heuristic instead.

New ticket for finding a heuristic for this: https://github.com/ROCm/rocMLIR-internal/issues/1587

**Comment by manupak** [comment]
I ve done MI300X - SDXL conv analysis. I think its worthwhile having the split-k = 3 solution.
We can defer anything clever to be done to the next ticket.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/mkarunar/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/mkarunar/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


type | f | I | O | n | c | H | W | K | y | x | p,q,u,v,l,j,g | PerfConfig-old | ConvTime-old | PerfConfig0-new | ConvTime-new | Diff Pct
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
convfp16 | GN01C | N01GC | N01GC | 2 | 1280 | 32 | 32 | 1280 | 3 | 3 | 1 | 128,128,8,32,32,8,3,1,1 | 0.207952 | 64,64,8,32,32,8,4,1,1 | 0.218158 | -1.0206
convfp16 | GN01C | N01GC | N01GC | 2 | 320 | 128 | 128 | 320 | 3 | 3 | 1 | 64,128,8,64,32,8,1,1,1 | 0.206421 | 128,128,8,64,16,8,1,1,1 | 0.210189 | -0.3768
convfp16 | GN01C | N01GC | N01GC | 2 | 320 | 128 | 128 | 320 | 3 | 3 | 1 | 128,128,8,128,32,8,1,1,1 | 0.0681314 | 128,128,8,128,32,8,1,1,1 | 0.0678018 | 0.03296
convfp16 | GN01C | N01GC | N01GC | 2 | 320 | 128 | 128 | 4 | 3 | 3 | 1 | 16,64,8,16,16,8,1,1,1 | 0.0354658 | 16,64,8,16,16,8,1,1,1 | 0.0357049 | -0.02391
convfp16 | GN01C | N01GC | N01GC | 2 | 320 | 64 | 64 | 640 | 3 | 3 | 1 | 128,256,8,64,32,8,1,1,1 | 0.109903 | 128,256,8,64,32,8,1,1,1 | 0.104875 | 0.5028
convfp16 | GN01C | N01GC | N01GC | 2 | 4 | 128 | 128 | 320 | 3 | 3 | 1 | 32,128,4,32,16,8,1,1,1 | 0.0433313 | 32,128,4,32,16,8,1,1,1 | 0.0444061 | -0.10748
convfp16 | GN01C | N01GC | N01GC | 2 | 640 | 128 | 128 | 320 | 3 | 3 | 1 | 128,128,4,64,32,8,1,1,1 | 0.365622 | 128,128,4,64,32,8,1,1,1 | 0.343494 | 2.2128
convfp16 | GN01C | N01GC | N01GC | 2 | 640 | 128 | 128 | 640 | 3 | 3 | 1 | 128,128,8,32,32,8,1,1,1 | 0.607927 | 128,128,8,128,32,8,1,1,1 | 0.608697 | -0.077
convfp16 | GN01C | N01GC | N01GC | 2 | 1280 | 64 | 64 | 1280 | 3 | 3 | 1 | 128,128,8,128,32,8,4,1,1 | 0.687461 | 128,128,8,128,32,8,4,1,1 | 0.68648 | 0.0981
convfp16 | GN01C | N01GC | N01GC | 2 | 640 | 32 | 32 | 1280 | 3 | 3 | 1 | 64,64,4,64,32,8,1,1,1 | 0.14516 | 128,128,8,128,32,8,1,1,1 | 0.122488 | 2.2672
convfp16 | GN01C | N01GC | N01GC | 2 | 640 | 64 | 64 | 640 | 3 | 3 | 1 | 128,128,4,128,32,8,1,1,1 | 0.230478 | 128,128,4,128,32,8,1,1,1 | 0.226099 | 0.4379
convfp16 | GN01C | N01GC | N01GC | 2 | 640 | 64 | 64 | 640 | 3 | 3 | 1 | 64,64,8,32,32,8,1,1,1 | 0.0823655 | 64,64,8,32,32,8,1,1,1 | 0.0805414 | 0.18241
convfp16 | GN01C | N01GC | N01GC | 2 | 960 | 128 | 128 | 320 | 3 | 3 | 1 | 128,128,4,32,32,8,1,1,1 | 0.53306 | 128,128,4,64,32,8,1,1,1 | 0.511213 | 2.1847
convfp16 | GN01C | N01GC | N01GC | 2 | 960 | 64 | 64 | 640 | 3 | 3 | 1 | 128,128,4,64,16,8,3,1,1 | 0.301662 | 128,128,4,32,32,8,1,1,1 | 0.303588 | -0.1926
convfp16 | GN01C | N01GC | N01GC | 2 | 1280 | 64 | 64 | 640 | 3 | 3 | 1 | 128,256,4,64,16,8,3,1,1 | 0.380245 | 128,128,8,128,32,8,4,1,1 | 0.425653 | -4.5408
convfp16 | GN01C | N01GC | N01GC | 2 | 1920 | 32 | 32 | 1280 | 3 | 3 | 1 | 128,128,8,32,32,8,3,1,1 | 0.283514 | 128,128,8,32,32,8,4,1,1 | 0.31805 | -3.4536
convfp16 | GN01C | N01GC | N01GC | 2 | 1920 | 64 | 64 | 640 | 3 | 3 | 1 | 128,128,8,32,32,8,4,1,1 | 0.51443 | 128,128,8,32,32,8,4,1,1 | 0.507413 | 0.7017
convfp16 | GN01C | N01GC | N01GC | 2 | 2560 | 32 | 32 | 1280 | 3 | 3 | 1 | 128,128,8,128,32,8,3,1,1 | 0.350949 | 64,64,8,32,32,8,4,1,1 | 0.418671 | -6.7722



</body>

</html>

**Review by manupak — CHANGES_REQUESTED** [review]
Lets make the range : 1,3,4.

**Review by manupak — APPROVED** [review]

**Pull request merged**